# Lab 1: Exploratory Data Analysis

by Nino Castellano, Will Lahners, Eddie Powers

### 1. Business Understanding

*** eddies part***

============================

[Dataset: Automobile Sales | Visual EDA](https://www.kaggle.com/code/ddosad/automobile-sales-visual-eda)

Can we identify any seasonal trends in toy sales, and how can this information be used for inventory planning? 

### 2. Data Understanding

##### 2.1 Data Description

In [2]:
#Required Libraries
import numpy as np
import pandas as pd

import seaborn as sns
import matplotlib.pyplot as plt
import plotly.graph_objects as go
import plotly.express as px

from warnings import filterwarnings
filterwarnings('ignore')

In [7]:
#Loading the file
df = pd.read_csv("https://raw.githubusercontent.com/ncastellano2320/Machine-Learning-in-Python/86d705d3789e7cbfeb3ecabcbd34709c116b456c/Lab1_ExploringTableData/Auto%20Sales%20data.csv?token=GHSAT0AAAAAACNGPIZJTKC4KS2NBQD6INVAZN5MMQQ", parse_dates=['ORDERDATE'], dayfirst=True)

#Replace the deal size to represent small, medium, and large with "1", "2" or "3" respectively. 
df.DEALSIZE.replace(to_replace = ['Small', 'Medium', 'Large'],
                 value = range(1,4), inplace = True)

#Drop unnecessary data
df = df.drop(['PHONE', 'CUSTOMERNAME', 'CONTACTFIRSTNAME', 'CONTACTLASTNAME', 'ADDRESSLINE1', 'POSTALCODE', 'STATUS', 'ORDERLINENUMBER', 'DAYS_SINCE_LASTORDER', 'ORDERNUMBER', 'PRODUCTCODE'], axis=1)

#Splitting the ORDERDATE column into three separate columns: year, month, and day
df['YEAR'] = df['ORDERDATE'].dt.year
df['MONTH'] = df['ORDERDATE'].dt.month
df['DAY'] = df['ORDERDATE'].dt.day

#Dropping the original ORDERDATE column
df = df.drop(['ORDERDATE'], axis=1)

df.head()

QUANTITYORDERED  PRICEEACH    SALES  PRODUCTLINE  MSRP        CITY COUNTRY  \
0               30      95.70  2871.00  Motorcycles    95         NYC     USA   
1               34      81.35  2765.90  Motorcycles    95       Reims  France   
2               41      94.74  3884.34  Motorcycles    95       Paris  France   
3               45      83.26  3746.70  Motorcycles    95    Pasadena     USA   
4               36      96.66  3479.76  Motorcycles    95  Burlingame     USA   

   DEALSIZE  YEAR  MONTH  DAY  
0         1  2018      2   24  
1         1  2018      5    7  
2         2  2018      7    1  
3         2  2018      8   25  
4         2  2018     10   28

Not only did we drop some unnecessary data, but we also changed the data type of the "DEALSIZE" variable, as it makes more sense to be processed as integers to represent small, medium, and large with "1", "2" or "3", respectively. 

In [4]:
#Find the data type
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2747 entries, 0 to 2746
Data columns (total 9 columns):
 #   Column           Non-Null Count  Dtype         
---  ------           --------------  -----         
 0   QUANTITYORDERED  2747 non-null   int64         
 1   PRICEEACH        2747 non-null   float64       
 2   SALES            2747 non-null   float64       
 3   ORDERDATE        2747 non-null   datetime64[ns]
 4   PRODUCTLINE      2747 non-null   object        
 5   MSRP             2747 non-null   int64         
 6   CITY             2747 non-null   object        
 7   COUNTRY          2747 non-null   object        
 8   DEALSIZE         2747 non-null   int64         
dtypes: datetime64[ns](1), float64(2), int64(3), object(3)
memory usage: 193.3+ KB


Based on the above output of the dataframe, there are no missing values within our dataset as every column's non-null count adds up to the total number of rows in the data, 2747. 

Because there are no missing rows, it would be unnecessary to do any imputation on the data, as it is a complete dataset. 

In [5]:
#Describe the data
df.describe()

QUANTITYORDERED    PRICEEACH         SALES  \
count      2747.000000  2747.000000   2747.000000   
mean         35.103021   101.098952   3553.047583   
min           6.000000    26.880000    482.130000   
25%          27.000000    68.745000   2204.350000   
50%          35.000000    95.550000   3184.800000   
75%          43.000000   127.100000   4503.095000   
max          97.000000   252.870000  14082.800000   
std           9.762135    42.042549   1838.953901   

                           ORDERDATE         MSRP     DEALSIZE  
count                           2747  2747.000000  2747.000000  
mean   2019-05-13 21:56:17.211503360   100.691664     1.601747  
min              2018-01-06 00:00:00    33.000000     1.000000  
25%              2018-11-08 00:00:00    68.000000     1.000000  
50%              2019-06-24 00:00:00    99.000000     2.000000  
75%              2019-11-17 00:00:00   124.000000     2.000000  
max              2020-05-31 00:00:00   214.000000     3.000000  
std                              NaN    40.114802     0.591981

In [6]:
#More detailed descriptions of the data
data_des = pd.DataFrame()

data_des['Features'] = df.columns
data_des['Description'] = ['order number', 'how many items within the order',
                          'the price at which the items are sold',
                          'total revenue from the order', 'date the order was submitted',
                          'days since coustomer\'s last order', 'manufacturer\'s suggested retail price',
                          'size of the deal']
# data_des['Scales'] = ['nominal'] * 1 + ['ratio'] * 3 +['interval']
# data_des['Discrete\Continuous'] = ['continuous'] * 2 + ['discrete'] + \
#                                   ['continuous'] + ['discrete'] * 6
# data_des['Range'] = ['0.09 - 1.00', '0.36 - 1.00', '2 - 7', '96 - 310',
#                     '2 - 10'] + ['0: No; 1: Yes'] * 3 + [repr(set(df.department))] + \
#                     ['1: low; 2: medium; 3: high']

data_des


ValueError: Length of values (8) does not match length of index (9)